shell/common/variable_refresh_rate_display.cc

```diff
@@ -9,17 +9,20 @@ namespace flutter {
 
 VariableRefreshRateDisplay::VariableRefreshRateDisplay(
     DisplayId display_id,
-    const VariableRefreshRateReporter& refresh_rate_reporter)
-    : Display(display_id, refresh_rate_reporter.GetRefreshRate()),
+    const std::weak_ptr<VariableRefreshRateReporter> refresh_rate_reporter)
+    : Display(display_id, refresh_rate_reporter.lock()->GetRefreshRate()),
```

Comment:
no nullptr check because we should always assume the `refresh_rate_reporter` is not null during construction. Same below

Reply:
`refresh_rate_reporter` might not be null but `refresh_rate_reporter.lock()` could be if the primary owner of the reporter is destructed during the construction of this object.